src/caffe/parallel.cpp

```diff
@@ -82,14 +83,17 @@ GPUParams<Dtype>::GPUParams(shared_ptr<Solver<Dtype> > root_solver, int device)
 
   // Allocate device buffers
   CUDA_CHECK(cudaSetDevice(device));
-  CUDA_CHECK(cudaMalloc(&data_, size_ * sizeof(Dtype)));
+  buffer_device_ = device;
+  GPUMemoryManager::allocate(reinterpret_cast<void **>(&data_),
```

Comment:
I see... the documentation declared `cudaMalloc` simply has the `void**` type, but of course there's more to it than that. It's also perhaps a bit surprising that `Dtype*` implicitly converts to `void*`, but `Dtype**` does not implicitly convert to `void**`... A wrapper seems reasonable.